src/com/xilinx/rapidwright/rwroute/RouteNode.java

```diff
@@ -432,7 +371,70 @@ public short getEndTileYCoordinate() {
      * @return The base cost of a RouteNode Object.
      */
     public float getBaseCost() {
-        return baseCost;
+        float _baseCost = 0;
```

Comment:
I would be curious to benchmark this properly -- we're potentially adding multiple branches, `getIntentCode()`, and floating point multiply to this.

Another approach I can think of is to have a `float[][]` array where the first dimension is `TileTypeEnum` and the second is `wireIndex` in order to get the base cost. Assuming the cost truly is the same for all tile types. Perhaps the same could be done for `length` too, assuming the same invariant applies.

Reply:
In the follow up to this optimization, I've found that this this method is called less than once per node pushed.  
```
Nodes pushed:                        605400311
Nodes popped:                        278681998
getBaseCost() calls:                 604688664
```
I'm open to optimizing it further.  For example, `node.getAllDownhillNodes().isEmpty()` can be precomputed offline and cached.